### PR TITLE
[WIP] Enable aliases names to contain the dot character

### PR DESCRIPTION
**Pull request by clarmso**
I have removed the constraint that the aliases' names cannot contain dots.  The issue is in querying.coffee.  The change is to prevent the characters at and after the dot being removed.

Addresses #3512

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
This isn't the most well documented feature, but when referencing aliases, you can reference @alias.all or @alias.2 and it will return respectively all of the aliases of that name or the 2nd request (index?) matching the name.

We want to continue supporting this API if the PR can be updated to support these uses.

**Comment by jennifer-shehane** [comment]
I've added a new issue to document this .all and .1 behavior in our documentation here: https://github.com/cypress-io/cypress-documentation/issues/1573

**Comment by clarmso** [comment]
@jennifer-shehane I'm on it. Thanks for clarifying the behaviour of the alias.all and alias.[number].

**Comment by brian-mann** [comment]
@jennifer-shehane the `.all` and `.1` is purposefully undocumented because it's behavior has never been specified. It's incomplete, inconsistent, and experimental. Not sure we really want to document it else when we change it it will be a breaking change.

**Comment by clarmso** [comment]
@jennifer-shehane Please note that the test failure from commands/xhr_spec.coffee is present in the develop branch as well.  It is failing consistently on Chrome 73 but passing on Electron 59 on my computer.

File: packages/driver/test/cypress/integration/commands/querying_spec.coffee [hunk 1]
```diff
@@ -883,6 +883,16 @@ describe "src/cy/commands/querying", ->
             .get("@getUsers").then (xhr) ->
               expect(xhr.url).to.include "/users"
 
+        it "handles dots in alias name", ->
```
Comment (jennifer-shehane) [inline packages/driver/test/cypress/integration/commands/querying_spec.coffee]: This test ends up throwing an error - when attempting to define an alias with a dot in its name for a route. I think if we will support dots in alias names, it should support it in routes as well as DOM els.

Screen Shot 2019-05-07 at 12 43 02 PM

Comment (clarmso) [reply packages/driver/test/cypress/integration/commands/querying_spec.coffee]: I've added some code to handle the dot character in the route name. Are there any other kind of aliases that I should be aware?

**Comment by clarmso** [comment]
Bleh! I deleted my fork without thinking about this PR. :( I'm still on it. I may open a new PR cherry-picking the changes.

**Comment by clarmso** [comment]
The work has been transplanted to https://github.com/cypress-io/cypress/pull/4248

**Pull request closed without merging**